implementations/rust/ockam/ockam_command/src/util/github.rs

```diff
@@ -0,0 +1,130 @@
+use std::{fs::File, io::Write, path::PathBuf};
+
+use clap::crate_version;
+use colorful::Colorful;
+use miette::{miette, IntoDiagnostic};
+use serde::Deserialize;
+use tokio::runtime::Builder;
+
+use crate::error::Result;
+
+#[derive(Deserialize, Debug)]
+pub struct LatestRelease {
+    name: String,
+}
+
+impl LatestRelease {
+    pub fn version(&self) -> Result<&str> {
+        let result = self.name.split_once('v');
+        match result {
+            Some((_, version)) => Ok(version),
+            None => Err(miette!("Failed to get latest release version").into()),
+        }
+    }
+}
+
+pub fn get_latest_release_version_sync() -> Result<LatestRelease> {
+    Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .unwrap()
+        .block_on(get_latest_release_version())
+}
+
+pub async fn get_latest_release_version() -> Result<LatestRelease> {
+    let url = "https://api.github.com/repos/build-trust/ockam/releases/latest";
+    let client = reqwest::Client::new();
+    let resp = client
+        .get(url)
+        .header("Accept", "application/vnd.github+json")
+        .header("X-GitHub-Api-Version", "2022-11-28")
+        .header("User-Agent", "Ockam Command")
+        .send()
+        .await;
+    if let Ok(r) = resp {
+        if let Ok(release) = r.json::<LatestRelease>().await {
+            return Ok(release);
+        }
+    }
+    Err(miette!("Failed to get latest release").into())
```

Comment:
It would be good to at least log the reason why the latest release could not be accessed by logging the error response.